src/dft.cpp

```diff
@@ -1375,4 +1375,91 @@ void fields::get_mode_mode_overlap(void *mode1_data, void *mode2_data, dft_flux
   get_overlap(mode1_data, mode2_data, flux, 0, overlaps);
 }
 
+std::vector<int> fields::get_corners(dft_fields fdft, component c){ // get the minimum and maximum ivec values of the dft monitor
+  volume *where = &v;
+  ivec min_corner = gv.round_vec(where->get_max_corner()) + one_ivec(gv.dim);
+  ivec max_corner = gv.round_vec(where->get_min_corner()) - one_ivec(gv.dim);
+
+  for (dft_chunk *chunk = fdft.chunks; chunk; chunk = chunk->next_in_dft) {
+    if (chunk->c != c) continue;
+    ivec isS = chunk->S.transform(chunk->is, chunk->sn) + chunk->shift;
+    ivec ieS = chunk->S.transform(chunk->ie, chunk->sn) + chunk->shift;
+    min_corner = min(min_corner, min(isS, ieS));
+    max_corner = max(max_corner, max(isS, ieS));
+  }
+  am_now_working_on(MpiAllTime);
+  max_corner = max_to_all(max_corner);
+  min_corner = -max_to_all(-min_corner); // i.e., min_to_all
+  finished_working();
+
+  std::vector<int> corners = {min_corner.x(),min_corner.y(),min_corner.z(),max_corner.x(),max_corner.y(),max_corner.z()};
+  return corners;
+}
+
+std::vector<struct sourcedata> dft_fields::fourier_sourcedata(const volume &where, int* min_max_corners, std::complex<double>* dJ){
```

Comment:
```suggestion
std::vector<struct sourcedata> dft_fields::fourier_sourcedata(const volume &where, const int* min_max_corners, const std::complex<double>* dJ){
```
since these arrays aren't modified?  (and change the headers etc. accordingly)

Reply:
They are not modified. `const` can be added here and also in `near_sourcedata` of `near2far.cpp`.